control-plane-operator/hostedclusterconfigoperator/controllers/resources/konnectivity/reconcile.go

```diff
@@ -73,10 +73,14 @@ func ReconcileAgentDaemonSet(daemonset *appsv1.DaemonSet, deploymentConfig confi
 			},
 		},
 	}
-	if platform != hyperv1.IBMCloudPlatform {
-		daemonset.Spec.Template.Spec.HostNetwork = true
+	if platform.Type != hyperv1.IBMCloudPlatform {
 		// Default is not the default, it means that the kubelets will re-use the hosts DNS resolver
 		daemonset.Spec.Template.Spec.DNSPolicy = corev1.DNSDefault
+		daemonset.Spec.Template.Spec.HostNetwork = true
+	} else {
+		if platform.IBMCloud != nil && platform.IBMCloud.ProviderType != configv1.IBMCloudProviderTypeUPI {
+			daemonset.Spec.Template.Spec.DNSPolicy = corev1.DNSDefault
+		}
```

Comment:
For code readability, I think it will be better to set the defaults in the DaemonSet spec and override specific values for IBM Cloud UPI.

Reply:
@hasueki are you saying that `corev1.DNSDefault` should be the default in the DaemonSet spec and we should override DNSPolicy here and only for IBM Cloud UPI?  I think that is reasonable.

Reply:
Yes, it is reasonable. I wanted to keep the changes low, but you are right. It looks much better. Please take a look. Thanks!